src/main/java/BlueTurtle/summarizers/Summarizer.java

```diff
@@ -29,6 +32,34 @@ public Summarizer(String packageName) {
 		setPackageName(packageName);
 		setWarningTypes(new HashSet<String>());
 		setNumberOfWarnings(0);
+		setNumberOfCheckStyleWarnings(0);
+		setNumberOfPMDWarnings(0);
+		setNumberOfFindBugsWarnings(0);
+	}
+
+	/**
+	 * Increment the number of warnings.
+	 * 
+	 * @param type
+	 *            the type of the warning.
+	 */
+	public void incrementNumberOfWarnings(String type) {
+		switch (type) {
+		case "CheckStyle":
+			numberOfCheckStyleWarnings++;
+			numberOfWarnings++;
+			break;
+		case "PMD":
+			numberOfPMDWarnings++;
+			numberOfWarnings++;
+			break;
+		case "FindBugs":
+			numberOfFindBugsWarnings++;
+			numberOfWarnings++;
+			break;
+		default:
+			numberOfWarnings += 0; // nothing should be added.
```

Comment:
The SAT gives a warning when there is no default case. Here the default case should never be triggered, so it's good practice to throw some kind of error if this happens.